[CI] ClientYamlTestSuiteIT test {yaml=indices.create/20_synthetic_source/object array in object with dynamic override} failing

**Build Scans:**
- [elasticsearch-intake #10490 / rest-compat](https://gradle-enterprise.elastic.co/s/q6rtpkkbmr2wm)
- [elasticsearch-intake #10489 / rest-compat](https://gradle-enterprise.elastic.co/s/3rruxo5wirsk6)
- [elasticsearch-pull-request #34098 / rest-compatibility](https://gradle-enterprise.elastic.co/s/eaz3ei5vryp7s)
- [elasticsearch-pull-request #34052 / rest-compatibility](https://gradle-enterprise.elastic.co/s/jzejiyguy55ag)
- [elasticsearch-pull-request #33988 / rest-compatibility](https://gradle-enterprise.elastic.co/s/q6j7ge2gfnbuo)

**Reproduction Line:**
```
./gradlew ":rest-api-spec:yamlRestCompatTest" --tests "org.elasticsearch.test.rest.ClientYamlTestSuiteIT" -Dtests.method="test {yaml=indices.create/20_synthetic_source/object array in object with dynamic override}" -Dtests.seed=9930015B2071D0B1 -Dtests.locale=guz-KE -Dtests.timezone=America/Miquelon -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Byaml%3Dindices.create%2F20_synthetic_source%2Fobject%20array%20in%20object%20with%20dynamic%20override%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.test.rest.ClientYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [indices.create/20_synthetic_source:714]: field [hits.hits.0._source.path_no.0.some_int] is null
```

**Issue Reasons:**
- [main] 2 consecutive failures in step rest-compat
- [main] 2 consecutive failures in pipeline elasticsearch-intake
- [main] 5 failures in test test {yaml=indices.create/20_synthetic_source/object array in object with dynamic override} (1.4% fail rate in 353 executions)
- [main] 2 failures in step rest-compat (8.3% fail rate in 24 executions)
- [main] 3 failures in step rest-compatibility (3.0% fail rate in 101 executions)
- [main] 2 failures in pipeline elasticsearch-intake (7.4% fail rate in 27 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (2.9% fail rate in 105 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -344,9 +344,7 @@ tests:
 - class: org.elasticsearch.kibana.KibanaThreadPoolIT
   method: testBlockedThreadPoolsRejectUserRequests
   issue: https://github.com/elastic/elasticsearch/issues/113939
-- class: org.elasticsearch.test.rest.ClientYamlTestSuiteIT
-  method: test {yaml=indices.create/20_synthetic_source/object array in object with dynamic override}
-  issue: https://github.com/elastic/elasticsearch/issues/113962
+
 
 # Examples:
 #

